invenio_rdm_records/services/access/service.py

```diff
@@ -385,6 +387,14 @@ def bulk_create_grants(self, identity, id_, data, expand=False, uow=None):
             raise GrantExistsError()
 
         for grant in grants:
+            # checks if groups are enabled in the instance
+            if grant["subject"]["type"] == RecordAccessService.group_subject_type:
```

Comment:
Groups functionality is depending on `USERS_RESOURCES_GROUPS_ENABLED` feature flag.
`can_groups_enabled` permission is managing it. Added checks for each service method to not allow users playing with groups through API if feature is disabled.